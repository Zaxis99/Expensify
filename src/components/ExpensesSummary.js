import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    return (
        <h1>Viewing {expenseCount} {expenseCount > 1 ? 'expenses': 'expense'} totalling {numeral(expenseTotal / 100).format('$0,0.00')}</h1>
    );
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);