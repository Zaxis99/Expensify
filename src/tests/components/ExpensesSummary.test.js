import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should correctly render expenses summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expesnseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expesnseCount={2} expensesTotal={235123456}/>);
    expect(wrapper).toMatchSnapshot();
});