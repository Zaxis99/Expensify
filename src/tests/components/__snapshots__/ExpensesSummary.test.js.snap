// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<h1>
  Viewing 
   
  expense
   totalling 
  $0.00
</h1>
`;

exports[`Should correctly render expenses summary with multiple expense 1`] = `
<h1>
  Viewing 
   
  expense
   totalling 
  $0.00
</h1>
`;
